src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -962,6 +962,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <PropertyGroup>
       <PublishedSingleFileName>$(AssemblyName)$(_NativeExecutableExtension)</PublishedSingleFileName>
+      <PublishDir Condition="!HasTrailingSlash('$(PublishDir)')">$(PublishDir)\</PublishDir>
```

Comment:
Why is `PrepareForPublish` not being called? Which target is being called ? `GenerateSingleFileBundle`? 
I think not calling `PrepareForPublish` can create other issues like: 
* Not checking for error cases (ex: trying to create a single-file for a class-library) which may cause unpredictable failures.
*  Not creating the `PublishDir`. 

Can we make the target getting called depend on PrepareForPublish?

Reply:
@swaroop-sridhar  This is for the wapproj scenario, the wapproj targets use ComputeFilesToPublish, which depends on GenerateSingleFileBundle.

If you don't want me to add this check here how about I just add PrepareForPublish to the ComputeFilesToPublish depends on list? (right above GenerateSingleFileBundle)

Reply:
Yes, I think it is better to add `PrepareForPublish` as a dependency.

Reply:
Sounds good, updated